### PR TITLE
PLANNER-753: Workbench: Score type selector is not fully visible in planner data object editor

### PR DESCRIPTION
**Pull request by mcimbora**
Replaced complex bootstrap select by errai-based select.

@ge0ffrey Can you please take a look? I've created 7.0.x issue for this as well, it's an annoying issue from usability perspective.

### CODE REVIEW

**Comment by mcimbora** [comment]
7.0.x version of https://github.com/kiegroup/optaplanner-wb/pull/157

**Comment by mcimbora** [comment]
Hi @manstis, can you please take a look at the PR? Thank you in advance.

File: optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditorViewImpl.java [hunk 1]
```diff
@@ -168,8 +177,18 @@ public boolean isFieldPickerEmpty() {
...
-        UIUtil.initList( planningSolutionScoreTypeSelector, options, selectedScoreType, false );
+    public void initPlanningSolutionScoreTypeOptions(List<Pair<String, String>> optionPairs) {
+        for (Pair<String, String> optionPair : optionPairs) {
+            Option option = createOption(optionPair);
+            planningSolutionScoreTypeSelector.add(option);
+        }
+    }
+
+    private Option createOption(final Pair<String, String> optionPair) {
+        Option option = (Option) document.createElement("option");
```
Comment (manstis) [inline optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditorViewImpl.java:188]: Out of interest.... open the editor, close it, open it, close it, open it, close it.... examing the DOM. Are spare ```option``` elements (I don't think there are as ```planningSolutionScoreTypeSelector``` would be removed from the DOM and all it's children.. but whenever I see creation without destruction I like to ask).

Comment (mcimbora) [reply optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/PlannerDataObjectEditorViewImpl.java:188]: @manstis The option list is initialized only once during object editor bean initialization. A good point anyway - in my first implementation I hit exactly the same problem you described.

**Review by manstis — APPROVED** [review]
Looks fine with me; I guess @ge0ffrey will want to handle merging to OptaPlanner repositories.

**Pull request merged**